.github/workflows/ci_cd.yml

```diff
@@ -59,88 +65,101 @@ jobs:
   doc-build:
     name: "Documentation building"
     needs: [style, doc-style]
+    timeout-minutes: 60
+    outputs:
+      PYMAPDL_VERSION: ${{ steps.version.outputs.PYMAPDL_VERSION }}
     runs-on: ubuntu-latest
     env:
       PYMAPDL_PORT: 21000  # default won't work on GitHub runners
       PYMAPDL_DB_PORT: 21001  # default won't work on GitHub runners
       PYMAPDL_START_INSTANCE: FALSE
-      DOCKER_PACKAGE: ghcr.io/ansys/mapdl
-      DOCKER_IMAGE_VERSION_DOCS_BUILD: v23.1.0
-      DPF_PORT: 21002
-
+      ON_DOCUMENTATION: TRUE
+    
     steps:
       - name: "Install Git and checkout project"
         uses: actions/checkout@v4
+
+      - name: "Login in Github Container registry"
+        uses: docker/login-action@v3.1.0
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: "Pull, launch, and validate MAPDL service"
+        id: start_mapdl
+        env:
+          LICENSE_SERVER: ${{ secrets.LICENSE_SERVER }}
+          MAPDL_VERSION: ${{ env.MAPDL_IMAGE_VERSION_DOCS_BUILD }}
+          DISTRIBUTED_MODE: "dmp"
+        run: |
+          export INSTANCE_NAME=MAPDL_0
+          .ci/start_mapdl.sh & export DOCKER_PID=$!
+          echo "Launching MAPDL service at PID: $DOCKER_PID"
+          echo "DOCKER_PID=$(echo $DOCKER_PID)" >> $GITHUB_OUTPUT
+
+      - name: "DPF Server Activation"
+        run: |
+          $(docker pull ghcr.io/ansys/dpf-core:22.2dev && docker run -d --name dpfserver -p ${{ env.DPF_PORT }}:50052 ghcr.io/ansys/dpf-core:22.2dev && echo "DPF Server active on port ${{ env.DPF_PORT }}.") &
+
+      - name: "Install OS packages"
+        run: |
+          sudo apt-get update
+          sudo apt install zip pandoc libgl1-mesa-glx xvfb texlive-latex-extra latexmk graphviz texlive-xetex libgomp1
      
       - name: "Set up Python using cache"
         uses: actions/setup-python@v5
         with:
           python-version: ${{ env.MAIN_PYTHON_VERSION }}
           cache: 'pip'
           cache-dependency-path: 'requirements/requirements_doc.txt'
-
-      - name: "Install system dependencies"
-        run: |
-          sudo apt-get update
-          sudo apt install zip pandoc libgl1-mesa-glx xvfb texlive-latex-extra latexmk graphviz texlive-xetex libgomp1
-
+        
       - name: "Install Python requirements"
-        run: pip install -r requirements/requirements_doc.txt
+        run: |
+          pip uninstall ansys-mapdl-core
```

Comment:
```suggestion
```

Reply:
As we are using the main branch of PyMAPDL and as it changes frequently, it is needed if we use the Python cache. I will add an if statement to be more precise

Reply:
As we set up Python with a cache and as we want to run the CICD with the `main` branch of PyMAPDL (which frequently changes), I would keep it - especially for the PyAnsys Math `main` branch.